drivers/storage/portworx/cloud_storage.go

```diff
@@ -98,6 +105,46 @@ func (p *portworxCloudStorage) GetStorageNodeConfig(
 	), nil
 }
 
+func (p *portworxCloudStorage) CreateStorageDistributionMatrix() error {
+	cm := &v1.ConfigMap{}
+	err := p.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      storageDecisionMatrixCMName,
+			Namespace: p.namespace,
+		},
+		cm,
+	)
+	if err == nil {
+		// config map already exists
+		return nil
+	}
+	yamlParser := parser.NewStorageDecisionMatrixParser()
+	matrixFileName := path.Join(specDir, string(p.cloudProvider)+".yaml")
+	matrix, err := yamlParser.UnmarshalFromYaml(matrixFileName)
+	if err != nil {
+		return err
+	}
+	yamlBytes, err := yamlParser.MarshalToBytes(matrix)
+	if err != nil {
+		return err
+	}
+
+	decisionMatrixCM := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      storageDecisionMatrixCMName,
+			Namespace: p.namespace,
+		},
+		Data: map[string]string{
+			storageDecisionMatrixCMKey: string(yamlBytes),
+		},
+	}
+	return p.k8sClient.Create(
```

Comment:
set owner references so that the yaml gets deleted when the operator is deleted.